src/main/cpp/src/get_json_object.hpp

```diff
@@ -23,6 +23,265 @@
 
 namespace spark_rapids_jni {
 
+namespace detail {
+/**
+ * JSON generator
+ */
+template <int max_json_nesting_depth = curr_max_json_nesting_depth>
+class json_generator {
+ public:
+  CUDF_HOST_DEVICE json_generator(char* _output, size_t _output_len)
+    : output(_output), output_len(_output_len)
+  {
+  }
+  CUDF_HOST_DEVICE json_generator() : output(nullptr), output_len(0) {}
+
+  enum class item { OBJECT, ARRAY, EMPTY };
+  // create a nested child generator based on this parent generator
+  // child generator is a view
+  CUDF_HOST_DEVICE json_generator new_child_generator()
+  {
+    if (nullptr == output) {
+      return json_generator();
+    } else {
+      return json_generator(output + output_len, 0);
+    }
+  }
+
+  CUDF_HOST_DEVICE json_generator finish_child_generator(json_generator const& child_generator)
+  {
+    // logically delete child generator
```

Comment:
What does this mean? We do that at the caller?

Reply:
Thanks Nghia, I think this PR is out of scope now as parser maintains context and implements `copy_current_structure`. I'll have this PR taken down when the PR #1868 is merged. 

Are we using this function `finish_child_generator` in the new implementation? @res-life

Reply:
>Are we using this function finish_child_generator in the new implementation? @res-life

No.